universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java

```diff
@@ -47,7 +59,8 @@ public ProgramServiceImpl(ProgramRepository programRepository) {
   public CompletionStage<Optional<ProgramDefinition>> getProgramDefinitionAsync(long id) {
     return programRepository
         .lookupProgram(id)
-        .thenApply(optionalProgram -> optionalProgram.map(Program::getProgramDefinition));
+        .thenApply(
+            optionalProgram -> optionalProgram.map((p) -> syncQuestions(p.getProgramDefinition())));
```

Comment:
issue (blocking): this version of `syncQuestions` makes a blocking call to the DB which block whatever thread calls `getProgramDefinitionAsync`

Reply:
it is nonblocking now